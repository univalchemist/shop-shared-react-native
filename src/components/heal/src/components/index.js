import { StackBackButton, healNavigationOptions } from './NavigationButtons';
import NonTouchableSearchBar from './NonTouchableSearchBar';
import DoctorCard from './DoctorCard';
export { default as SpinningLoader } from './SpinningLoader';

export {
  DoctorCard,
  StackBackButton,
  healNavigationOptions,
  NonTouchableSearchBar,
};
