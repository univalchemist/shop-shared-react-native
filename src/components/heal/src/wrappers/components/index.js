import React from 'react';

export {
  Image,
  Text,
  PlainText,
  Box,
  Button,
  Divider,
  Flex,
  SectionHeadingText,
  ErrorPanel,
  ListItemWithRightChevron,
  ScrollView,
  Icon,
  Loader,
} from '@cxa-rn/components';
export {
  withTracking,
  TrackedButton,
  TrackedFloatingActionButton,
  TrackedListItem,
  TrackedCarousel,
  TrackedCarouselWithScrollBar,
} from './Tracking';
