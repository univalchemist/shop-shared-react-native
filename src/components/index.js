/* istanbul ignore file */

export { default as DocumentViewerModal } from './DocumentViewerModal';
export { default as WalletBalancePanel } from './WalletBalancePanel';
export { default as SvgText } from './SvgText';
