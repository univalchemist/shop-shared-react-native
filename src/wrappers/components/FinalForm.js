export { InputField, SelectField } from '@cxa-rn/components';
