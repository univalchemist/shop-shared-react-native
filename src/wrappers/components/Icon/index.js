import React from 'react';
import { Icon as CommonIcon } from '@cxa-rn/components';
import { createIconSetFromIcoMoon } from 'react-native-vector-icons';
import icoMoonConfig from '@config/icomoon.json';
const IcoMoon = createIconSetFromIcoMoon(icoMoonConfig);

const Icon = props => {
  const { icomoon } = props;
  const IconCom = icomoon ? IcoMoon : CommonIcon;
  return <IconCom {...props} />;
};

export default Icon;
