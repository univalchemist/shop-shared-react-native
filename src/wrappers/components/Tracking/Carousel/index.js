import React, { useState } from 'react';
import { Carousel, withScrollBar } from '@cxa-rn/components';
import { useDebouncedCallback } from 'use-debounce';
import { logAction } from '@store/analytics/trackingActions';

export const TrackedCarousel = props => {
  const { actionParams, onSnapToItem, data } = props;
  const [maxReachedIndex, setMaxReachedIndex] = useState(0);

  const [debounceLogAction] = useDebouncedCallback(percent => {
    logAction({
      ...actionParams,
      percent,
    });
  }, 1000 * 30);

  return (
    <Carousel
      {...props}
      onSnapToItem={snapIndex => {
        if (onSnapToItem) {
          onSnapToItem(snapIndex);
        }

        if (actionParams && snapIndex > maxReachedIndex) {
          setMaxReachedIndex(snapIndex);

          debounceLogAction((snapIndex + 1) / data.length);
        }
      }}
    />
  );
};

export const TrackedCarouselWithScrollBar = withScrollBar(TrackedCarousel);

export default Carousel;
