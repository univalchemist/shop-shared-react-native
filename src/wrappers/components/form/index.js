import React from 'react';
import { CustomMultiselectCheckBox as MultiselectCheckBox } from '@cxa-rn/components';
import { TrackedButton } from '../Tracking';

export {
  formFocusField as focusField,
  InputFieldWithoutScroll as InputField,
  SelectFieldWithoutScroll as SelectField,
  RadioButtonGroupWithoutScroll as RadioButtonGroup,
  QuestionText,
  CheckBoxField,
} from '@cxa-rn/components';
export { default as SingleSelectModal } from './SingleSelectModal';
export { default as MultiSelectModal } from './MultiSelectModal';

export const CustomMultiselectCheckBox = props => {
  return <MultiselectCheckBox {...props} TrackedButton={TrackedButton} />;
};

// import FeatureToggle from '@config/FeatureToggle';
// import MultiselectCheckBox, {
//   CustomMultiselectCheckBox,
// } from './MultiselectCheckBox';

// let RadioButtonGroup = FeatureToggle.NO_DEFAULT_ANSWERS.on
//   ? RadioButtonGroupWithScroll
//   : RadioButtonGroupWithoutScroll;

// let SelectField = FeatureToggle.NO_DEFAULT_ANSWERS.on
//   ? SelectFieldWithScroll
//   : SelectFieldWithoutScroll;

// let InputField = FeatureToggle.NO_DEFAULT_ANSWERS.on
//   ? InputFieldWithScroll
//   : InputFieldWithoutScroll;
