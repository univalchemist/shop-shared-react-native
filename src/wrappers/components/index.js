import React from 'react';
import { ListPicker as DefaultListPicker } from '@cxa-rn/components';
import { TrackedListItemPicker } from './Tracking';

export {
  Box,
  Button,
  Container,
  Datepicker,
  ScrollView,
  StepProgressBar,
  Text,
  SectionHeadingText,
  Image,
  ListItem,
  DocumentUploader,
  DocumentUploaderWithScrollBar,
  LabelValueText,
  ScreenHeadingText,
  SecondaryText,
  TextSkeletonPlaceholder,
  ListSkeletonPlaceholder,
  ImageSkeletonPlaceholder,
  SectionListSkeletonPlaceholder,
  StatusPanel,
  FieldSkeletonPlaceholder,
  ButtonSkeletonPlaceholder,
  ErrorPanel,
  ListContainer,
  FormattedMoney,
  IconSkeletonPlaceholder,
  Loader,
  PlainText,
  Footer,
  UploadBox,
  StackBackButton,
  ListItemContainer,
  Flex,
  SvgSkeletonPlaceholder,
  SkeletonPlaceholder,
  Card,
  SvgText,
  ErrorText,
  ImageErrorCard,
  CardLoader,
  Divider,
  ListItemWithRightChevron,
  TouchableContainer,
  ListHeader,
  SectionLabelValueListSkeletonPlaceholder,
  ProgressBar,
  AppVersion,
  ParagraphSkeletonPlaceholder,
  HeaderButton,
  ModalBackButton,
  ScrollViewForStickyButton,
  Input,
  Carousel,
} from '@cxa-rn/components';
export {
  withTracking,
  TrackedButton,
  TrackedFloatingActionButton,
  TrackedListItem,
  TrackedCarousel,
  TrackedCarouselWithScrollBar,
} from './Tracking';

export { default as Icon } from './Icon';

export const ListPicker = props => {
  return (
    <DefaultListPicker TrackedListItem={TrackedListItemPicker} {...props} />
  );
};
