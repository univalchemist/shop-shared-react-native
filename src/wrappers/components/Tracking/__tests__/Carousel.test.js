import React from 'react';
import { renderForTest } from '@testUtils';
import { TrackedCarousel } from '../Carousel';
import { Carousel } from '@cxa-rn/components';
import { Text } from 'react-native';

const data = [
  {
    title: 'Item 1',
    text: 'Text 1',
  },
  {
    title: 'Item 2',
    text: 'Text 2',
  },
];

describe('TrackedCarousel', () => {
  it('should call snapToItem and logAction', () => {
    const component = renderForTest(
      <TrackedCarousel
        inactiveSlideScale={1}
        inactiveSlideOpacity={1}
        data={data}
        renderItem={({ item }) => <Text>{item.text}</Text>}
        sliderWidth={300}
        itemWidth={300}
        onSnapToItem={jest.fn()}
        actionParams={{
          category: 'LIFESTYLE_OVERVIEW',
          action: 'Horizontal scroll of more lifestyle tips',
        }}
      />,
    );

    const carousel = component.queryAllByType(Carousel);
    carousel[0].props.onSnapToItem(1);
    expect(carousel.length).toEqual(1);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should not call snapToItem and logAction', () => {
    const component = renderForTest(
      <TrackedCarousel
        inactiveSlideScale={1}
        inactiveSlideOpacity={1}
        data={data}
        renderItem={({ item }) => <Text>{item.text}</Text>}
        sliderWidth={300}
        itemWidth={300}
      />,
    );

    const carousel = component.queryAllByType(Carousel);
    carousel[0].props.onSnapToItem(2);
    expect(carousel.length).toEqual(1);
  });
});
