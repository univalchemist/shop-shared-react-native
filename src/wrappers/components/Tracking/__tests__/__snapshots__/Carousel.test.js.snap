// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackedCarousel should call snapToItem and logAction 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTScrollView
                        actionParams={
                          Object {
                            "action": "Horizontal scroll of more lifestyle tips",
                            "category": "LIFESTYLE_OVERVIEW",
                          }
                        }
                        activeAnimationOptions={null}
                        activeAnimationType="timing"
                        activeSlideAlignment="center"
                        activeSlideOffset={20}
                        apparitionDelay={0}
                        automaticallyAdjustContentInsets={false}
                        autoplay={false}
                        autoplayDelay={1000}
                        autoplayInterval={3000}
                        callbackOffsetMargin={5}
                        containerCustomStyle={Object {}}
                        contentContainerCustomStyle={Object {}}
                        contentContainerStyle={
                          Array [
                            Object {
                              "paddingLeft": 0,
                              "paddingRight": 0,
                            },
                            Object {},
                          ]
                        }
                        data={
                          Array [
                            Object {
                              "text": "Text 1",
                              "title": "Item 1",
                            },
                            Object {
                              "text": "Text 2",
                              "title": "Item 2",
                            },
                          ]
                        }
                        decelerationRate="fast"
                        directionalLockEnabled={true}
                        disableVirtualization={false}
                        enableMomentum={false}
                        enableSnap={true}
                        firstItem={0}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        hasParallaxImages={false}
                        horizontal={true}
                        inactiveSlideOpacity={1}
                        inactiveSlideScale={1}
                        inactiveSlideShift={0}
                        initialNumToRender={6}
                        invertStickyHeaders={false}
                        inverted={false}
                        itemWidth={300}
                        keyExtractor={[Function]}
                        layout="default"
                        lockScrollTimeoutDuration={1000}
                        lockScrollWhileSnapping={false}
                        loop={false}
                        loopClonesPerSide={3}
                        maxToRenderPerBatch={13}
                        numColumns={1}
                        onContentSizeChange={[Function]}
                        onEndReachedThreshold={2}
                        onLayout={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onResponderRelease={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        onSnapToItem={[Function]}
                        onStartShouldSetResponderCapture={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        overScrollMode="never"
                        pinchGestureEnabled={false}
                        removeClippedSubviews={true}
                        renderItem={[Function]}
                        scrollEnabled={true}
                        scrollEventThrottle={1}
                        scrollsToTop={false}
                        shouldOptimizeUpdates={true}
                        showsHorizontalScrollIndicator={false}
                        showsVerticalScrollIndicator={false}
                        slideStyle={Object {}}
                        sliderWidth={300}
                        stickyHeaderIndices={Array []}
                        style={
                          Object {
                            "flexDirection": "row",
                            "opacity": 0,
                            "width": 300,
                          }
                        }
                        swipeThreshold={20}
                        updateCellsBatchingPeriod={50}
                        useScrollView={false}
                        vertical={false}
                        viewabilityConfigCallbackPairs={Array []}
                        windowSize={13}
                      >
                        <View>
                          <View
                            onLayout={[Function]}
                            style={
                              Array [
                                Object {
                                  "flexDirection": "row",
                                },
                                null,
                              ]
                            }
                          >
                            <View
                              pointerEvents="box-none"
                              style={
                                Array [
                                  Object {
                                    "width": 300,
                                  },
                                  Object {},
                                  Object {},
                                ]
                              }
                            >
                              <Text>
                                Text 1
                              </Text>
                            </View>
                          </View>
                          <View
                            onLayout={[Function]}
                            style={
                              Array [
                                Object {
                                  "flexDirection": "row",
                                },
                                null,
                              ]
                            }
                          >
                            <View
                              pointerEvents="box-none"
                              style={
                                Array [
                                  Object {
                                    "width": 300,
                                  },
                                  Object {},
                                  Object {},
                                ]
                              }
                            >
                              <Text>
                                Text 2
                              </Text>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
